src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -2120,6 +2120,35 @@ public void test_submodule_checkout_simple() throws Exception {
         assertSubmoduleContents(w.repo);
     }
 
+    /* Opening a git repository in a directory with a symbolic git file instead
+     * of a git directory should function properly. JGit does not implement
+     * renamed submodules, so we the submodules branch won't work with it...
+     */
+    @NotImplementedInJGit
```

Comment:
I think this test can be implemented for JGit as well as for CliGit.  See the next comment for my assumption of how it can be implemented for JGit.